File: apps/plant_shop_c-sharp/plant_shop_c-sharp/Utils/PasswordUtil.cs
namespace plant_shop_c_sharp.Utils
{
    /// <summary>
    /// Utilitaire de hachage BCrypt.
    /// </summary>
    public static class PasswordUtil
    {
        /// <summary>
        /// Hache un mot de passe avec BCrypt.
        /// </summary>
        /// <param name="password">Mot de passe en clair.</param>
        /// <returns>Hash BCrypt.</returns>
        public static string HashPassword(string password)
        {
            return BCrypt.Net.BCrypt.HashPassword(password);
        }

        /// <summary>
        /// Verifie un mot de passe contre son hash.
        /// </summary>
        /// <param name="password">Mot de passe en clair.</param>
        /// <param name="hashedPassword">Hash BCrypt.</param>
        /// <returns>True si valide.</returns>
        public static bool CheckPassword(string password, string hashedPassword)
        {
            try
            {
                return BCrypt.Net.BCrypt.Verify(password, hashedPassword);
            }
            catch
            {
                return false;
            }
        }
    }
}

File: apps/plant_shop_java/Structure_en_application_distribuée/plant_shop_java_distribuée/utils/PasswordUtil.java
package util;

import org.mindrot.jbcrypt.BCrypt;

/**
 * Utilitaires pour le hachage et la vérification de mots de passe avec BCrypt.
 */
public final class PasswordUtil {

    /**
	 * Constructeur privé pour empêcher l'instanciation.
	 */
	private PasswordUtil() {}

    /**
	 * Hache un mot de passe en clair avec BCrypt.
	 * 
	 * @param plainTextPassword String Le mot de passe en clair
	 * @return String Le hash BCrypt du mot de passe
	 */
    public static String hashPassword(String plainTextPassword) {
        return BCrypt.hashpw(plainTextPassword, BCrypt.gensalt());
    }

    /**
	 * Vérifie qu un mot de passe en clair correspond à un hash BCrypt.
	 * 
	 * @param plainTextPassword String Le mot de passe en clair
	 * @param hashedPassword String Le hash BCrypt à vérifier
	 * @return boolean true si le mot de passe correspond, false sinon
	 */
    public static boolean checkPassword(String plainTextPassword, String hashedPassword) {
        if (plainTextPassword == null || hashedPassword == null) {
            return false;
        }
        try {
            return BCrypt.checkpw(plainTextPassword, hashedPassword);
        } catch (IllegalArgumentException e) {
            System.err.println("Avertissement : hash BCrypt invalide.");
            return false;
        }
    }
}
